pkg/controller/volume/persistentvolume/index.go

```diff
@@ -117,9 +117,14 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 		//   the claim.
 		for _, volume := range volumes {
 			if isVolumeBoundToClaim(volume, claim) {
-				// this claim and volume are bound; return it,
-				// whether the claim is prebound or for volumes
-				// intended for dynamic provisioning v1
+				// this claim and volume are pre-bound; return
+				// the volume if the size request is satisfied,
+				// otherwise leave the claim pending
+				volumeQty := volume.Spec.Capacity[api.ResourceStorage]
+				volumeSize := volumeQty.Value()
+				if volumeSize < requestedSize {
+					return nil, nil
```

Comment:
If you are pre-bound you don't get to choose another volume.

On Wed, Aug 17, 2016 at 1:25 AM, Jan Šafránek <notifications@github.com>
wrote:

> In pkg/controller/volume/persistentvolume/index.go
> <https://github.com/kubernetes/kubernetes/pull/30690#discussion_r75081112>
> :
>
> > @@ -117,9 +117,14 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
> >  		//   the claim.
> >  		for _, volume := range volumes {
> >  			if isVolumeBoundToClaim(volume, claim) {
> > -				// this claim and volume are bound; return it,
> > -				// whether the claim is prebound or for volumes
> > -				// intended for dynamic provisioning v1
> > +				// this claim and volume are pre-bound; return
> > +				// the volume if the size request is satisfied,
> > +				// otherwise leave the claim pending
> > +				volumeQty := volume.Spec.Capacity[api.ResourceStorage]
> > +				volumeSize := volumeQty.Value()
> > +				if volumeSize < requestedSize {
> > +					return nil, nil
>
> If the pre-bound PV is too small we should check all the other volumes and
> not blindly return here. There may be other PVs that could satisfy the
> claim.
>
> continue would be more appropriate.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/30690/files/fe817674abac9a7a2a8ac53d9f2debae9c9d36be#r75081112>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVLTGMQ0jiADNen4QysuW7wmQQ6mUks5qgsVugaJpZM4Jlm5v>
> .
>